src/OS8104A.ts

```diff
@@ -0,0 +1,697 @@
+import { Gpio } from "onoff"
+import spi, { type SpiDevice, type SpiOptions } from "spi-device"
+import EventEmitter from "events"
+import { getRegisterConfig } from "./RegisterConfig"
+import { Registers } from "./Registers.js"
+const TRANSFER_SPEED = 180000
+
+export interface MostMessage {
+    targetAddressHigh: number
+    targetAddressLow: number
+    fBlockID: number
+    instanceID: number
+    fktId: number
+    opType: number
+    data: number[]
+}
+
+export interface Config {
+    freq: number
+    nodeAddressLow: number
+    nodeAddressHigh: number
+    groupAddress: number
+}
+
+export interface Stream {
+    sourceAddrHigh: number
+    sourceAddrLow: number
+    fBlockID: number
+    instanceID: number
+    sinkNr: number
+}
+
+export interface AllocResult {
+    loc1: number
+    loc2: number
+    loc3: number
+    loc4: number
+    cl: number
+    answer1: string
+    freeChannels: number
+}
+
+export interface SourceResult {
+    nodePos: number
+    group: number
+    logicalHigh: number
+    logicalLow: number
+}
+
+const options: SpiOptions = {
+    chipSelectHigh: false,
+    bitsPerWord: 8,
+    lsbFirst: false
+}
+
+export class OS8104A extends EventEmitter {
+    readonly freq: number
```

Comment:
You could define a frequency type - limiting to specific allowed numeric value with a union type:

`type Frequency = 1 | 2 | 3`

Reply:
frequency removed as actually unused